app/controllers/application_controller.rb

```diff
@@ -17,7 +17,8 @@ class ApplicationController < Sinatra::Base
   end
 
   post '/registrations' do
-    @user = User.find_by(email: params[:email])
```

Comment:
Since the '/registrations' controller action block receives information from the signup form, it will not be able to find a User who has just registered.